app_legacy/Helpers/database/site-award.php

```diff
@@ -58,56 +58,47 @@ function getUsersWithAward(int $awardType, int $data, ?int $dataExtra = null): a
 
 function getUsersSiteAwards(string $user, bool $showHidden = false): array
 {
-    sanitize_sql_inputs($user);
-
-    $retVal = [];
+    $dbResult = [];
 
     if (!isValidUsername($user)) {
-        return $retVal;
+        return $dbResult;
     }
 
+    $bindings = [
+        'username' => $user,
+        'username2' => $user,
+    ];
+
     $query = "
-    (
-    SELECT UNIX_TIMESTAMP( saw.AwardDate ) as AwardedAt, saw.AwardType, saw.AwardData, saw.AwardDataExtra, saw.DisplayOrder, gd.Title, c.Name AS ConsoleName, gd.Flags, gd.ImageIcon
+    SELECT " . unixTimestampStatement('saw.AwardDate', 'AwardedAt') . ", saw.AwardType, saw.AwardData, saw.AwardDataExtra, saw.DisplayOrder, gd.Title, c.Name AS ConsoleName, gd.Flags, gd.ImageIcon
                   FROM SiteAwards AS saw
                   LEFT JOIN GameData AS gd ON ( gd.ID = saw.AwardData AND saw.AwardType = " . AwardType::Mastery . " )
                   LEFT JOIN Console AS c ON c.ID = gd.ConsoleID
-                  WHERE saw.AwardType = " . AwardType::Mastery . " AND saw.User = '$user'
+                  WHERE saw.AwardType = " . AwardType::Mastery . " AND saw.User = :username
                   GROUP BY saw.AwardType, saw.AwardData, saw.AwardDataExtra
-    )
     UNION
-    (
-    SELECT UNIX_TIMESTAMP(MAX( saw.AwardDate )) as AwardedAt, saw.AwardType, MAX( saw.AwardData ), saw.AwardDataExtra, saw.DisplayOrder, NULL, NULL, NULL, NULL
+    SELECT " . unixTimestampStatement('MAX(saw.AwardDate)', 'AwardedAt') . ", saw.AwardType, MAX( saw.AwardData ), saw.AwardDataExtra, saw.DisplayOrder, NULL, NULL, NULL, NULL
                   FROM SiteAwards AS saw
-                  WHERE saw.AwardType > " . AwardType::Mastery . " AND saw.User = '$user'
+                  WHERE saw.AwardType > " . AwardType::Mastery . " AND saw.User = :username2
                   GROUP BY saw.AwardType
-
-    )
     ORDER BY DisplayOrder, AwardedAt, AwardType, AwardDataExtra ASC";
 
-    $db = getMysqliConnection();
-    $dbResult = mysqli_query($db, $query);
+    $dbResult = legacyDbFetchAll($query, $bindings)->toArray();
```

Comment:
line 88 (`$dbResult !== false`) should always be true now. i'm surprised phpstan didn't complain

Reply:
Sounds like it might be best if I just remove that conditional. Would that be acceptable to do in this PR?

Reply:
Yes.

Reply:
Resolved in https://github.com/RetroAchievements/RAWeb/pull/1545/commits/01fcb2447ef3935edee585a5dd425d7869397972